### DISCUSSION TITLE
Alguém pode me ajudar aqui ?

### DISCUSSION THREAD

**Discussion by alissondias99** [frontendbr/forum#2225] [Perguntas e Respostas] [answered]
Olá, estou no inicio da caminhada do front-end, e estou tendo dificuldades com um exercício o dia todo.
Basicamente o exercício em questão tem que receber n números e em seguida mostrar qual é o maior, menor, a soma e a média, mas nada disso ocorre, eu já tentei reescrever o código, olhar códigos dos outros, mas nada funciona.
Se alguém puder dar um olhada.

HTML: 
<!DOCTYPE html>
<html lang="pt-br">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Analisador de números</title>
    <link rel="stylesheet" href="style.css">
</head>
<body>
    <header>
       Analisador de números 
    </header>

    <section>
        <div>

          Números entre 1 e 100: <input type="number" name="nuns" id="nuns"> <button onclick='adicionar()'>Adicionar</button> <br> <br>
          <select name="números" id="lista" size="10"></select> <br>
          <button onclick="finalizar()">Finalizar</button>
        </div>

        <div id=res>

        </div>

    </section>
    <footer>
        <p>Site criado por Alisson Dias </p>
    </footer>
    <script src="script.js"></script>
</body>
</html>

JS:
let nuns = document.getElementById('nuns')
let lista = document.getElementById('lista')
let result = document.createElement('res')
let valores = []

function isNumero(n) {
    if (Number(n) >= 1 && Number(n) <= 100){
        return true
    } else {
        return false
    }
}

function inLista(n, l) {
    if (l.indexOf(Number(n)) != -1){
        return true
    } else{
        return false
    }
}

function adicionar(){
    if(isNumero(nuns.value) && !inLista(nuns.value)){
        valores.push(Number(nuns.value))
        let item = document.createElement('option')
        item.text = `nuns ${nuns.value} adicionado` 
        lista.appendChild(item)
    } else {
        window.alert("nuns inválido ou já encontrado no lista.")
    }
    num.value =''
    num.focus()
}

function finalizar() {
    if (valores.length == 0) {
        window.alert("Adicione valores antes de continuar")
    } else {
        let tot = valores.length
        let maior = valores[0]
        let menor = valores[0]
        let soma = 0
        let media = 0
        for (let pos in valores) {
            soma += valores[pos]
            if(maior  = valores[pos]){
                maior = valores[pos]
            } else if(valores[pos] < menor) {
                menor = valores[pos]
            }
        }
        media = soma / tot
        res.innerHTML = ``
        res.innerHTML += `<p>Ao todo temos ${tot} números cadastrados.</p>`
        res.innerHTML += `<p> O maior valor informado foi ${maior}</p>`
        res.innerHTML += `<p> O menor valor informado foi ${menor}</p>`
        res.innerHTML += `<p> A soma de todos os valores informados é ${soma}</p>`
        res.innerHTML += `<p> A média dos valores informados é ${media}</p>`
    }
}

Essa é a mensagem de erro que aparece no navegador (Chrome)
Uncaught TypeError: Cannot read properties of undefined (reading 'indexOf')
    at inLista (script.js:15)
    at adicionar (script.js:23)
    at HTMLButtonElement.onclick (index.html:18)

**Comment by juanpinheiro** [comment, marked as answer]
Fala Alisson, tudo bem?

Sua função `inLista()` recebe depois parâmetros. Primeiro é o `n` referente ao valor e o `l` referente a lista (de acordo com o que eu entendi do seu código).

O erro acontece pois no seu if `if(isNumero(nuns.value) && !inLista(nuns.value))` você só está passando um único parâmetro pra essa função.

Olhando o seu código eu dirita que sua função deveria ficar mais ou menos assim:

```javascript
function adicionar() {
    if (isNumero(nuns.value) && !inLista(nuns.value, valores)) {
        valores.push(Number(nuns.value));
        let item = document.createElement('option');
        item.text = `nuns ${nuns.value} adicionado`;
        lista.appendChild(item);
    } else {
        window.alert("nuns inválido ou já encontrado no lista.");
    }
    nuns.value = ''; // num.value = ''; Você teria um outro erro nessa linha, pois a variável num não existe
    nuns.focus(); // num.focus(); O mesmo problema da linha anterior
}
```

Aparentemente só erros de digitação, continue estudando e entendendo o código que logo você vai estar fera!

Bons estudos, abraço!

**Reply by alissondias99** [reply]
Muito obrigado pela ajudo e pelo incentivo cara, me ajudo muito mesmo, pode deixar que eu não vou desanimar do vida de programador.
